Add wifi provisioning BLE service (#10178)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst

.. _wifi_prov_readme:

Wi-Fi Provisioning Service
##########################

.. contents::
   :local:
   :depth: 2

Overview
********

> This module implements of an Bluetooth GATT service for Wi-Fi provisioning.

Reviewer: ```suggestion
This module implements a Bluetooth® GATT service for Wi-Fi® provisioning.
```

The service is divided into three parts:

> * GATT service inerface: Defines GATT service and provide an interface for configurator to exchange messages.

Reviewer: ```suggestion
* GATT service inerface: Defines GATT service and provides an interface for the configurator to exchange messages.
```

  * If the command is ``START_SCAN``, the result message sent to the configurator contains information about the AP.
    Each ``Result`` contains information related to one AP.
  * If the command is ``SET_CONFIG``, when the Wi-Fi status changes (for example, from disconnected to connected), the configurator receives a result message with the new status.
    Meanwhile, the Wi-Fi credentials are stored in the non-volatile memory of the device.

See all definitions in :file:`./common/proto`.

Operations
==========

The message sequence is the same for all operations, with variations depending on the command:

1. The configurator sends an encoded ``Request`` message with a command to the device.
#. The device carries out the command.
#. The device sends a ``Response`` message to the configurator.
#. For some operations, the device also sends a ``Result`` message to the configurator with any additional data generated or reported.

See :ref:`wifi_provisioning_message_types` for more information on the commands and additional parameters.

Configuration management
************************

The configuration management component manages Wi-Fi configurations.

> It uses the :ref:`lib_wifi_credentials` to handle the configurations in flash.

Reviewer 2: ```suggestion
It uses the :ref:`lib_wifi_credentials` library to handle the configurations in flash.
```

> It uses the :ref:`lib_wifi_credentials` to handle the configurations in flash.

Reviewer 2: ```suggestion
It uses the :ref:`lib_wifi_credentials` library to handle the configurations in flash.
```
